fatiando/gridder.py

```diff
@@ -439,3 +445,281 @@ def cut(x, y, scalars, area):
+                       constant_values=(float(padtype), float(padtype)))
+    elif padtype.lower() == 'mean':
+        # Pad with the mean
+        ap = numpy.pad(a, nps, mode='mean')
+    elif padtype.lower() == 'lintaper':
+        # Linearly taper to the mean
+        ap = numpy.pad(a, nps, mode='linear_ramp', end_values=(m, m))
+    elif padtype.lower() == 'edge':
+        # Pad with edge values
+        ap = numpy.pad(a, nps, mode='edge')
+    elif padtype.lower() == 'reflection':
+        # Pad with even reflection
+        ap = numpy.pad(a, nps, mode='reflect',
+                       reflect_type='even')
+    elif padtype.lower() == 'oddreflection':
+        # Pad with odd reflection
+        ap = numpy.pad(a, nps, mode='reflect',
+                       reflect_type='odd')
+    elif padtype.lower() == 'oddreflectiontaper':
+        # Pad with odd reflection and a cosine taper to mean
+        ap = (numpy.pad(a, nps, mode='reflect',
+                        reflect_type='odd') - m)
+        for ii in range(0, nd):
+            ap = numpy.apply_along_axis(_costaper, ii, ap, lp=nps[ii][0],
+                                        rp=nps[ii][1])
+        ap += m
+
+    return ap, nps
+
+
+def unpad_array(a, nps):
+    '''
+    Unpads an array using the outputs from pad_array.
+
+    This function takes a padded array and removes the padding from both.
+    Effectively, this is a complement to gridder.cut for when you already
+    know the number of elements to remove.
+
+    .. note:: Unlike :func:`~fatiando.gridder.pad_array`, this returns a slice
+        of the input array.  Therefore, any changes to the padded array will be
+        reflected in the unpadded array.
+
+    Parameters:
+
+    * a : N-D array
+        Array to be un-padded.  Can be of arbitrary dimension.
+    * nps : list
+        List of tuples giving the min and max indices for the cutoff.
+        Identical to nps returned by pad_array
+
+    Returns:
+
+    * b : N-D array
+        Array of same dimension as a, with padding removed
+
+    '''
+    o = []
+    for ii in range(0, a.ndim):
+        o.append(slice(nps[ii][0], a.shape[ii] - nps[ii][1]))
+    b = a[o]
+
+    return b
+
+
+def padcoords(xy, s, nps):
+    '''
+    Pads coordinate vectors.
+
+    Designed to be used in concert with :func:`~fatiando.gridder.pad_array`,
+    this function takes a list of coordinate vectors and pads them using the
+    same discretization.
+
+    Parameters:
+
+    * xy : List
+        List of arrays of coordinates
+    * s : tuple
+        Size of original array array
+    * nps : tuple
+        Size of padded array
+
+    Returns:
+
+    * coordspad : List
+        List of padded coordinate arrays
+
+    '''
+    coords = []
+    d = []
+    coordspad = []
+    for ii in range(0, len(s)):
+        if type(xy) is numpy.ndarray:
+            coords.append(xy)
+        elif type(xy) is list and len(s) > 1:
+            coords.append(xy[ii].reshape(s).transpose().take(0, axis=ii))
```

Comment:
type checking in Python is not a very good idea because it invalidates all the possible [duck typing](https://en.wikipedia.org/wiki/Duck_typing) that makes Python so flexible. For example, I could not  pass an [xray](http://xray.readthedocs.org/en/stable/) array to this function but there is no reason why it shouldn't work.

A better option in this case is to ask for a list of 1d arrays and assume that that is what you get.  So you can do:

    for array in xy: